### PR TITLE
SDIT-1593: ✨ More switching to authentication holder

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.kt [hunk 1]
```diff
@@ -316,7 +317,7 @@ class AppointmentsService(
   }
 
   private fun findLocationInUserLocations(appointmentLocation: AgencyInternalLocation): Optional<Location> {
-    val userLocations = locationService.getUserLocations(authenticationFacade.currentPrincipal, true)
+    val userLocations = locationService.getUserLocations(hmppsAuthenticationHolder.username, true)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AppointmentsService.kt:320]: switch to `username` as must be a nomis user to get locations

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -677,7 +674,7 @@ public List<OffenderSentenceDetail> getOffenderSentencesSummary(final String age
     }
 
     public List<OffenderSentenceDetail> getBookingSentencesSummary(final List<Long> bookingIds) {
-        final var offenderSentenceSummary = bookingSentenceSummaries(bookingIds, caseLoadService.getCaseLoadIdsForUser(authenticationFacade.getCurrentPrincipal(), false),
+        final var offenderSentenceSummary = bookingSentenceSummaries(bookingIds, caseLoadService.getCaseLoadIdsForUser(hmppsAuthenticationHolder.getUsername(), false),
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:677]: switch to `username` as must be a nomis user to get case loads

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 3]
```diff
@@ -884,7 +881,7 @@ private void validateUpdateLivingUnit(final OffenderBooking offenderBooking, fin
     }
 
     private Set<String> getCaseLoadIdForUserIfRequired() {
-        return hasAnySystemOverrideRole(RESTRICTED_ALLOWED_ROLES) ? Set.of() : caseLoadService.getCaseLoadIdsForUser(authenticationFacade.getCurrentPrincipal(), false);
+        return hasAnySystemOverrideRole(RESTRICTED_ALLOWED_ROLES) ? Set.of() : caseLoadService.getCaseLoadIdsForUser(hmppsAuthenticationHolder.getUsername(), false);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:884]: switch to `username` as must be a nomis user to get case loads

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 4]
```diff
@@ -972,7 +969,7 @@ private List<OffenderSentenceDetailDto> offenderSentenceSummaries(final String a
     }
 
     private List<OffenderSentenceDetailDto> offenderSentenceSummaries(final String agencyId, final Set<String> caseloads, final boolean filterByCaseloads) {
-        final var query = buildAgencyQuery(agencyId, authenticationFacade.getCurrentPrincipal());
+        final var query = buildAgencyQuery(agencyId, hmppsAuthenticationHolder.getUsername());
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:972]: switch to `username` as must be a nomis user to get case loads (which the build eventually calls)

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 5]
```diff
@@ -1055,7 +1052,7 @@ public Page<PrisonerBookingSummary> getPrisonerBookingSummary(final String priso
             .prisonId(prisonId)
             .bookingSequence(1)
             .active(true)
-            .caseloadIds(viewAllPrisoners ? null : staffUserAccountRepository.getCaseloadsForUser(authenticationFacade.getCurrentPrincipal(), true, "INST").stream().map(Caseload::getId).toList())
+            .caseloadIds(viewAllPrisoners ? null : staffUserAccountRepository.getCaseloadsForUser(hmppsAuthenticationHolder.getUsername(), true, "INST").stream().map(Caseload::getId).toList())
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:1055]: switch to `username` as must be a nomis user to get case loads

File: src/main/java/uk/gov/justice/hmpps/prison/service/CaseLoadService.java [hunk 6]
```diff
@@ -19,15 +21,13 @@ public CaseLoadService(final CaseLoadRepository caseLoadRepository) {
         this.caseLoadRepository = caseLoadRepository;
     }
 
-    public Optional<CaseLoad> getCaseLoad(final String caseLoadId) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/CaseLoadService.java:22]: unused

File: src/main/java/uk/gov/justice/hmpps/prison/service/CaseLoadService.java [hunk 7]
```diff
@@ -19,15 +21,13 @@ public CaseLoadService(final CaseLoadRepository caseLoadRepository) {
...
-        return caseLoadRepository.getCaseLoad(caseLoadId);
-    }
-
     public List<CaseLoad> getCaseLoadsForUser(final String username, final boolean allCaseloads) {
+        if (StringUtils.isBlank(username)) return Collections.emptyList();
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/CaseLoadService.java:25]: short circuit - if we don't have a username then no point going to database

File: src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java [hunk 8]
```diff
@@ -169,11 +169,11 @@ public List<InmateBasicDetails> getBasicInmateDetailsForOffenders(final Set<Stri
     }
 
     private boolean isViewAllOffenders() {
-        return authenticationFacade.isOverrideRole("GLOBAL_SEARCH", "VIEW_PRISONER_DATA");
+        return hmppsAuthenticationHolder.isOverrideRole("GLOBAL_SEARCH", "VIEW_PRISONER_DATA");
     }
 
     private Set<String> loadCaseLoadsOrThrow() {
-        final var caseloads = caseLoadService.getCaseLoadIdsForUser(authenticationFacade.getCurrentPrincipal(), false);
+        final var caseloads = caseLoadService.getCaseLoadIdsForUser(hmppsAuthenticationHolder.getUsername(), false);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:176]: switch to `username` as must be a nomis user to get case loads

File: src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java [hunk 9]
```diff
@@ -523,7 +523,7 @@ public List<OffenderCategorise> getCategory(final String agencyId, final Categor
     @Transactional
     public Map<String, Long> createCategorisation(final Long bookingId, final CategorisationDetail categorisationDetail) {
         validate(categorisationDetail);
-        final var userDetail = userService.getUserByUsername(authenticationFacade.getCurrentPrincipal());
+        final var userDetail = userService.getUserByUsername(hmppsAuthenticationHolder.getUsername());
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:526]: switch to `username` as must be a nomis user to get user

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**